File: Lab-2/q3/app/src/main/java/com/example/q3/MainActivity.java
package com.example.q3;

import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button buttonRelative, buttonLinear;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        buttonRelative = findViewById(R.id.buttonrelative);
        buttonLinear = findViewById(R.id.buttonlinear);

        buttonRelative.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Start the RelativeActivity
                Intent intent = new Intent(MainActivity.this, MainActivityRelative.class);
                startActivity(intent);
            }
        });

        buttonLinear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Start the FirstActivity (assuming "first" is the name of the layout)
                Intent intent = new Intent(MainActivity.this, first.class);
                startActivity(intent);
            }
        });
    }
}

File: Lab-4/AndroidVersions/app/src/main/java/com/example/androidversions/MainActivity.kt
package com.example.androidversions

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.androidversionlist.VersionListFragment

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.container, VersionListFragment())
                .commit()
        }
    }
}
